public/components/GeoPlot.vue

```diff
@@ -427,8 +490,62 @@ export default Vue.extend({
     band(): string {
       return routeGetters.getBandCombinationId(this.$store);
     },
+    tileState(): MapState {
+      return {
+        onHover: (id: number) => {
+          if (id > this.areas.length) {
+            console.error(`id: ${id} is outside of this.areas bounds`);
+            return; // id outside of bounds
+          }
+          this.toastTitle = this.areas[id].imageUrl;
+          this.hoverItem = this.areas[id].item;
+          this.hoverUrl = this.areas[id].imageUrl;
+          this.$bvToast.show(this.toastId);
+          window.addEventListener("mousemove", this.fadeToast);
+        },
+        onClick: (id: number) => {
+          if (id > this.areas.length || id < 0) {
+            console.error(
+              `id retrieved from buffer picker ${id} not within index bounds of areas.`
+            );
+            return;
+          }
+          this.showImageDrilldown(this.areas[id].imageUrl, this.areas[id].item);
+        },
+        quads: () => {
+          return this.areaToQuads();
+        },
+      };
+    },
+    clusterState(): MapState {
+      return {
+        onHover: (id: number) => {}, // onHover empty for cluster state
+        onClick: (id: number) => {
+          if (id > this.bucketFeatures.length || id < 0) {
+            console.error(
+              `id retrieved from buffer picker ${id} not within index bounds of areas.`
+            );
+            return;
+          }
+          const bucket = this.bucketFeatures[id];
+          const point1 = this.renderer.latlngToNormalized(
+            bucket.coordinates[0]
+          );
+          const point2 = this.renderer.latlngToNormalized(
+            bucket.coordinates[1]
+          );
+          const center = {
+            x: (point1.x + point2.x) / 2,
+            y: (point1.y + point2.y) / 2,
+          };
+          this.map.zoomToPosition(this.zoomThreshold, center); // zoom to the center of the cluster clicked. Zoom to the point where the state switches
```

Comment:
Aren't you worried that on a smaller screen (i.e. laptop) that `this.zoomThreshold` might be too high and that the canvas might only show a partial bucket?